kong/runloop/wasm.lua

```diff
@@ -402,6 +402,8 @@ local function rebuild_state(db, version, old_state)
 
       for _, filter in ipairs(chain.filters) do
         if filter.enabled then
+          _M.filters_by_name[filter.name].config = cjson_decode(filter.config) or filter.config
+
```

Comment:
Proxy-wasm spec for metrics doesn't provide a way for filter developers to define a metric associated with a set of labels.

The function `proxy_define_metric` expects only a metric type and a string representing the metric name.

As a result, filter developers include whatever label associated with a metric as part of its name, e.g. `a_metric_label1="label_value"_label2="label_value"`.

There isn't however a consensus over how to represent labels in metric names.

The functionality of parsing labels proposed and implemented in the function `parse_pw_key` below, reflects the one available in Envoy and used by Coraza WAF Proxy-wasm filter.

It expects filter developers to provide a set of regex patterns matching labels in a metric's name, as part of the filter configuration:

```
    label_patterns = {
      { label = "service", pattern = "(_s_id=([0-9a-z%-]+))" },
      { label = "route", pattern = "(_r_id=([0-9a-z%-]+))" },
    }
```

Then, during serialization, those patterns are match with metric names to extract potential labels.

As such, we need to have access to the filter configuration when serializing Proxy-Wasm metrics.